en: update docs for tidb deployment (#17)
pingcap/docs-tidb-operator: en/deploy-on-general-kubernetes.md

- For the demonstration environment or functional verification, you can use network storage, such as `ebs` and `nfs`.

Different components of a TiDB cluster have different disk requirements. Before deploying a TiDB cluster, select the appropriate storage class for each component according to the storage classes supported by the current Kubernetes cluster and usage scenario.

You can set the storage class by modifying `storageClassName` of each component in `<cluster-name>/tidb-cluster.yaml` and `<cluster-name>/tidb-monitor.yaml`. For the [storage classes](configure-storage-class.md) supported by the Kubernetes cluster, check with your system administrator.

> **Note:**
>
> If you set a storage class that does not exist in the TiDB cluster that you are creating, then the cluster creation goes to the Pending state. In this situation, you must [destroy the TiDB cluster in Kubernetes](destroy-a-tidb-cluster.md).

### Cluster topology

The deployed cluster topology by default has 3 PD Pods, 3 TiKV Pods, and 2 TiDB Pods. In this deployment topology, the scheduler extender of TiDB Operator requires at least 3 nodes in the Kubernetes cluster to provide high availability.

> If the number of Kubernetes cluster nodes is less than 3, 1 PD Pod goes to the Pending state, and neither TiKV Pods nor TiDB Pods are created.

Reviewer: ```suggestion
If the number of Kubernetes cluster nodes is less than 3, 1 PD Pod will go to the Pending state, and neither TiKV Pods nor TiDB Pods will be created.
```

Author: @DanielZhangQD I suggest we use the present tense as much as possible and limit the use of "will". Please refer to [Present tense  |  Google developer documentation style guide](https://developers.google.cn/style/tense?hl=en)

Author: I suggest we use the present tense as much as possible and limit the use of "will". Please refer to [Present tense  |  Google developer documentation style guide](https://developers.google.cn/style/tense?hl=en)

When the number of nodes in the Kubernetes cluster is less than 3, to start the TiDB cluster, you can reduce both the number of PD Pods and TiKV Pods in the default deployment to `1`.

## Deploy TiDB Cluster

After you deploy and configure TiDB Operator, deploy the TiDB cluster by the following steps:

1. Create `Namespace`:

    {{< copyable "shell-regular" >}}

    ``` shell
    kubectl create namespace <namespace>
    ```

    > **Note:**
    >
    > A [`namespace`](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) is a virtual cluster backed by the same physical cluster. You can give it a name that is easy to memorize, such as the same name as `cluster-name`.

2. Deploy the TiDB cluster:

    {{< copyable "shell-regular" >}}

    ``` shell
    kubectl apply -f <cluster-name> -n <namespace>
    ```

3. View the Pod status:

    {{< copyable "shell-regular" >}}

    ``` shell
    kubectl get po -n <namespace> -l app.kubernetes.io/instance=<cluster-name>
    ```

> You can use TiDB Operator to deploy and manage multiple sets of TiDB clusters in a single Kubernetes cluster by repeating the above command and replacing `cluster-name` with a different name.

Reviewer: ```suggestion
You can use TiDB Operator to deploy and manage multiple TiDB clusters in a single Kubernetes cluster by repeating the above procedure and replacing `cluster-name` with a different name.
```
